include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -344,9 +637,402 @@ class alignment_file_format_sam
+    char buffer[317] = ""; // Doubles can be up to 316 characters
+
+    //!\brief A variable that tracks whether the header_ptr has been written or not.
     bool written_header{false};
 
+    /*!\brief Decays to detail::consume for std::ignore.
+     * \tparam stream_view_type  The type of the stream as a view.
+     * \tparam target_type       A type that decays to std::ignore.
+     *
+     * \param[in, out] stream_view  The stream view to consume.
+     * \param[in, out] target       A std::ignore placeholder.
+     */
+    template <typename stream_view_type, typename target_type>
+    //!\cond
+        requires detail::decays_to_ignore_v<target_type>
+    //!\endcond
+    void read_field(stream_view_type && stream_view, target_type SEQAN3_DOXYGEN_ONLY(target))
+    {
+        detail::consume(stream_view);
+    }
+
+    /*!\brief Reads a range by copying from stream_view to target, converting values with seqan3::view::char_to.
+     * \tparam stream_view_type  The type of the stream as a view.
+     * \tparam target_range_type The type of range to parse from input; Must model std::ranges::ForwardRange.
+     *
+     * \param[in, out] stream_view  The stream view to iterate over.
+     * \param[in, out] target       The range to store the parsed sequence.
+     */
+    template <typename stream_view_type, std::ranges::ForwardRange target_range_type>
+    //!\cond
+        requires requires (std::string s) { { view::char_to<value_type_t<target_range_type>>(s) }; }
+    //!\endcond
+    void read_field(stream_view_type && stream_view, target_range_type & target)
+    {
+        if (!is_char<'*'>(*std::ranges::begin(stream_view)))
+            ranges::copy(stream_view | view::char_to<value_type_t<target_range_type>>,
+                         ranges::back_inserter(target));
+        else
+            ++std::ranges::begin(stream_view); // skip '*'
+
+        assert(std::ranges::begin(stream_view) == std::ranges::end(stream_view)); // consumed everything
+    }
+
+    /*!\brief Reads arithmetic fields using std::from_chars.
+     * \tparam stream_view_type The type of the stream as a view.
+     * \tparam target_type      The type of value to parse from input; Must model seqan3::Arithmetic.
+     *
+     * \param[in, out] stream_view  The stream view to iterate over.
+     * \param[in, out] target       The arithmetic value object to store the parsed value.
+     *
+     * \throws seqan3::parse_error if the character sequence in stream_view cannot be successfully converted to a value
+     *         of type target_type.
+     */
+    template <typename stream_view_type, Arithmetic target_type>
+    void read_field(stream_view_type && stream_view, target_type & target)
+    {
+        // unfortunately std::from_chars only accepts char const * so we need a buffer.
+        auto end = std::ranges::copy(stream_view, buffer).out;
+        std::from_chars_result res = std::from_chars(&buffer[0], end, target);
```

Comment:
I don't see the advantage over a normal char array?

Reply:
It is a just safer in case some rouge developer is going to modify the code. It begins with using std::addressof instead of &-operator. And not getting the address of the tenth instead of the first value. So it is simply best practice to use safe interface over an unsafe c-style interface.